### PR TITLE
ISPN-12437 xsite max idle

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-12437

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
+Expiration removes cache entries based on time.
+{brandname} provides two ways to configure expiration for entries:
 
-* `lifespan` expiration is suitable for cross-site replication. When entries reach the maximum lifespan, {brandname} expires them independently of the remote sites.
+`lifespan`:: Sets the maximum amount of time that entries can exist. +
+When you set `lifespan` with cross-site replication, {brandname} clusters expire entries independently of remote sites.
 
-* `max-idle` expiration does not work with cross-site replication. {brandname} cannot determine when cache entries reach the idle timeout in remote sites.
+`max-idle`:: Specifies how long entries can exist based on read or write operations in a given time period. +
+When you set a `max-idle` with cross-site replication, {brandname} clusters send touch commands during replication to coordinate idle timeout with remote sites.
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc:10]: @wburns Does this description seem adequate? I didn't want to get overly complex about the touch commands. I know if they return null then the entry is expired. But I'm not sure we need to say that. 

Is there any performance penalty that comes from latency with touch commands?

Comment (pruivo) [reply documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc:10]: I believe the touch commands are sent on access/read. IMO, the word "replication" implies some kind of write operation and should be avoided.

Performance is expected to be worse. You will go from simple RAM access (best case) to a remote site round-trip (ouch!)... There is no free lunch here :) The good part is the touch command isn't required in every read...

File: documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+Expiration removes cache entries based on time.
+{brandname} provides two ways to configure expiration for entries:
 
-* `lifespan` expiration is suitable for cross-site replication. When entries reach the maximum lifespan, {brandname} expires them independently of the remote sites.
+`lifespan`:: Sets the maximum amount of time that entries can exist. +
+When you set `lifespan` with cross-site replication, {brandname} clusters expire entries independently of remote sites.
 
-* `max-idle` expiration does not work with cross-site replication. {brandname} cannot determine when cache entries reach the idle timeout in remote sites.
+`max-idle`:: Specifies how long entries can exist based on read or write operations in a given time period. +
+When you set a `max-idle` with cross-site replication, {brandname} clusters send touch commands on write operations, and some read operations, to coordinate idle timeout values with remote sites. +
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc:10]: I don't think we send touch on write.

**Review by pruivo — APPROVED** [review]

File: documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
+Expiration removes cache entries based on time.
+{brandname} provides two ways to configure expiration for entries:
 
-* `lifespan` expiration is suitable for cross-site replication. When entries reach the maximum lifespan, {brandname} expires them independently of the remote sites.
+`lifespan`:: Sets the maximum amount of time that entries can exist. +
+When you set `lifespan` with cross-site replication, {brandname} clusters expire entries independently of remote sites.
 
-* `max-idle` expiration does not work with cross-site replication. {brandname} cannot determine when cache entries reach the idle timeout in remote sites.
+`max-idle`:: Specifies how long entries can exist based on read or write operations in a given time period. +
+When you set a `max-idle` with cross-site replication, {brandname} clusters send touch commands to coordinate idle timeout values with remote sites. +
```
Comment (wburns) [inline documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc:10]: This is a bit generalized, but maybe that is the point? Seems fine though.

Comment (oraNod) [reply documentation/src/main/asciidoc/topics/con_xsite_expiration.adoc:10]: @wburns Yes it's a bit generalized but it seemed preferable to getting too low level. Do you think there are any points we should be more specific about or could we just merge this?

**Comment by wburns** [comment]
Integrated into master, thanks @oraNod !

**Pull request merged**